ui/client/datasets/DataTransformation/useElwoodData.js

```diff
@@ -16,61 +16,110 @@ const useElwoodData = ({
         `/api/dojo/job/${datasetId}/${jobString}`, requestArgs
       );
 
       if (jobQueueResp.status === 200) {
-        return jobQueueResp.data?.id;
+        return jobQueueResp.data;
       }
     };
 
-    const runElwoodJob = ({ requestArgs, onFailure }) => {
-      let count = 0;
-
-      const repeatFetch = (jobId) => {
-        setTimeout(() => {
-          axios.post(`/api/dojo/job/fetch/${jobId}`).then((response) => {
-            if (response.status === 200) {
-              // keep track of how long it takes (for dev purposes)
-              count += 1;
-              // console.log(`${count}: response from job string: ${jobString}:`, response);
-              if (response.data) {
-                console.log(`success! ${jobString} took ${count * 500}ms`, response.data);
-                // setOptions currently just used for onGeoResSuccess
-                onSuccess(response.data, setData, setDataError, setDataLoading, setOptions);
-                return;
-              }
-              if (cleanupRef.current) {
-                // if no data and the component is mounted, try the fetch again
-                repeatFetch(jobId);
-              }
+    // repeatedly calls /job/fetch/jobId until it gets the job results
+    const repeatFetch = (jobId, requestArgs, onFailure) => {
+      setTimeout(() => {
+        axios.post(`/api/dojo/job/fetch/${jobId}`).then((response) => {
+          if (response.status === 200) {
+            if (response.data) {
+              // setOptions currently just used for onGeoResSuccess
+              onSuccess(response.data, setData, setDataError, setDataLoading, setOptions);
+              return;
             }
-          }).catch((err) => {
-            // we get a 404 immediately - there is some sort of bug, this accounts for that
-            if (count < 2 && cleanupRef.current) {
-              // and the component is mounted
-              console.log('repeating job call');
-              startElwoodJob({ requestArgs }).then((resp) => {
-                repeatFetch(resp);
-              });
+            if (cleanupRef.current) {
+              // if no data and the component is mounted, try the fetch again
+              repeatFetch(jobId, requestArgs, onFailure);
+            }
+          }
+        }).catch(() => {
+          if (cleanupRef.current) {
+            // only do this if the parent component (cleanupRef) is still mounted
+            startElwoodJob({ requestArgs }).then((resp) => {
+              repeatFetch(resp, requestArgs, onFailure);
+            });
+          } else {
+            onFailure();
+          }
+        });
+      }, 500);
+    };
+
+    // the jobId is always this format so we can create it here
+    const jobId = `${datasetId}_${jobString}`;
+    // check if the job is already running or finished when we first hit the page
+    const checkExistingJob = async () => {
+      try {
+        const response = await axios.post(`/api/dojo/job/fetch/${jobId}`);
```

Comment:
We now will see an initial 404 the first time we hit the transformation page, however. That's because we need to check to see if the job is already running, and the very first time we get there it won't be running, so we'll get a 404 back, which tells us to kick it off. This is expected.